budgetportal/tests/test_prov_infra_chart.py

```diff
@@ -0,0 +1,817 @@
+        irm_snapshot_1 = IRMSnapshot.objects.create(
+            financial_year=self.fin_year,
+            quarter=q1,
+            date_taken=date(2030, 6, 30),
+            file=File(self.file),
+        )
+        ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=irm_snapshot_1,
+            project=self.project,
+            actual_expenditure_q1=10,
+            expenditure_from_previous_years_total=200,
+        )
+        q2 = Quarter.objects.create(number=2)
+        irm_snapshot_2 = IRMSnapshot.objects.create(
+            financial_year=self.fin_year,
+            quarter=q2,
+            date_taken=date(2030, 9, 30),
+            file=File(self.file_2),
+        )
+        ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=irm_snapshot_2,
+            project=self.project,
+            actual_expenditure_q1=11,
+            actual_expenditure_q2=20,
+            expenditure_from_previous_years_total=200,
+        )
+
+    def tearDown(self):
+        self.file.close()
+        self.file_2.close()
+        self.file_3.close()
+
+    def test_q1_q2_updated_after_q3_snapshot_inserted(self):
+        """Test that Q1 and Q2 are updated correctly when Q3 inserted"""
+        snapshots_data = json.loads(time_series_data(self.project))
+        snapshots_data = snapshots_data[u"snapshots"]
+        self.assertEqual(len(snapshots_data), 2)
+
+        # Check Q1 values
+        self.assertEqual(snapshots_data[0]["total_spent_in_quarter"], 11)
+        self.assertEqual(snapshots_data[0]["total_spent_to_date"], 211)
+
+        # Check Q2 values
+        self.assertEqual(snapshots_data[1]["total_spent_in_quarter"], 20)
+        self.assertEqual(snapshots_data[1]["total_spent_to_date"], 231)
+
+        # Create Q3 snapshot
+        q3 = Quarter.objects.create(number=3)
+        irm_snapshot_3 = IRMSnapshot.objects.create(
+            financial_year=self.fin_year,
+            quarter=q3,
+            date_taken=date(2030, 12, 31),
+            file=File(self.file_3),
+        )
+        ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=irm_snapshot_3,
+            project=self.project,
+            actual_expenditure_q1=12,
+            actual_expenditure_q2=21,
+            actual_expenditure_q3=30,
+            expenditure_from_previous_years_total=200,
+        )
+        # Recreate the chart data
+        snapshots_data = json.loads(time_series_data(self.project))
+        snapshots_data = snapshots_data[u"snapshots"]
+        self.assertEqual(len(snapshots_data), 3)
+
+        # Check Q1 values
+        self.assertEqual(snapshots_data[0]["total_spent_in_quarter"], 12)
+        self.assertEqual(snapshots_data[0]["total_spent_to_date"], 212)
+
+        # Check Q2 values
+        self.assertEqual(snapshots_data[1]["total_spent_in_quarter"], 21)
+        self.assertEqual(snapshots_data[1]["total_spent_to_date"], 233)
+
+
+class Q1Q2Q3UpdateTestCase(TransactionTestCase):
```

Comment:
please delete - I think the previous two are sufficient to prove the point

Reply:
Deleted.

https://github.com/vulekamali/datamanager/pull/215/commits/e48f74125c5e7c0745191f6801f5c0e1d70f9171